### ISSUE TITLE
💅 useNamingConvention: The rule should ignore variables that are destructured out of objects

### ISSUE THREAD

**Issue by vladkrasn** [biomejs/biome#2332]
### Rule name

useNamingConvention

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAYQAgAD0AIAB7AAoAIAAgAFcAcgBvAG4AZwBDAEEAUwBFAFMAVABSAEkAQwBUADoAIAAxACwACgAgACAAQQBMAFMATwBXAFIATwBOAEcAOgAgADEACgB9AAoACgBmAHUAbgBjAHQAaQBvAG4AIABmAHUAbgBjACAAKABhACkAIAB7AAoAYwBvAG4AcwB0ACAAewAgAFcAcgBvAG4AZwBDAEEAUwBFAFMAVABSAEkAQwBUACwAIABBAEwAUwBPAFcAUgBPAE4ARwAgAH0AIAA9ACAAYQAKAGMAbwBuAHMAdAAgAG8AbgBlACAAPQAgAEEATABTAE8AVwBSAE8ATgBHACAAKwAgADEACgBvAG4AZQAgACsAIABXAHIAbwBuAGcAQwBBAFMARQBTAFQAUgBJAEMAVAAgACsAIABBAEwAUwBPAFcAUgBPAE4ARwAKAH0ACgA%3D

### Expected result

Variables that are produced by destructured objects are not necessarily declared by a Biome user, they may come from a third party library. Maybe Biome should ignore them?

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
Instead of ignoring them, I think the expected behavior of this rule in this case should be suggesting renaming those destructured variables to comply with the naming convention.

**Comment by Conaclos** [comment]
I am hesitating between completely ignoring this kind of names or applying the suggestion of @Sec-ant.

**Comment by ematipico** [comment]
Ther's also the fact that there's a pattern to remove properties from an object, like this:
```js
const { SOMETHING, FOO, ...rest } = bar;

doSomething(rest);
```

This means that the user would be forced to write the code like this:

```js
const { SOMETHING: something, FOO: foo, ...rest } = bar;

doSomething(rest);
```

I think this is very user-hostile and doesn't provide good value to the rule.
